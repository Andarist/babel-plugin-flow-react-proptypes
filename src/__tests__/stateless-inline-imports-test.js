var babel = require('babel-core');
var content = `
// @flow

import React from 'react';
import type { T } from '../types';

const C = (props: T) => {
  <div>{props.name}</div>
};
`;

it('stateless-inline-imports', () => {
  var res = babel.transform(content, {
    babelrc: false,
    presets: ['es2015', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
