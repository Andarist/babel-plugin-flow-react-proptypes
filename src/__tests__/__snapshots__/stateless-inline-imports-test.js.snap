// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stateless-inline-imports 1`] = `
"'use strict';

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _types = require('../types');

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var C = function C(props) {
  _react2.default.createElement(
    'div',
    null,
    props.name
  );
};

C.propTypes = (_types.bpfrp_babelPluginFlowReactPropTypes_proptype_T || _propTypes2.default.func) === _propTypes2.default.any ? {} : _types.bpfrp_babelPluginFlowReactPropTypes_proptype_T || _propTypes2.default.func;"
`;
